server/authentication.js

```diff
@@ -8,42 +16,196 @@ function enforceHTTPS(request, response, next) {
   return next();
 }
 
+function sendUnauthorized(res) {
+  res.set('WWW-Authenticate', 'Basic realm=Authorization Required');
+  return res.send(401);
+}
+
+function getDomain(request) {
+  return (process.env.NODE_ENV === 'development')
+    ? 'http://localhost:5000'
+    : `https://${request.headers.host}`;
+}
+
 // Only allow folks who are authorized.  Call `next` if they are authorized,
 // and return a 404 with no body if not.
-function onlyAllowResearchers(request, response, next) {
+function onlyAllowResearchers(pool, request, response, next) {
   // TODO
   // 1. Read the token from the request header
   // 2. Check to see if the token header is in `tokens` database table and it's less than 24 hours old
   // 3. This code should run in both development and production.
+
   if (process.env.NODE_ENV === 'development') return next();
-  response.status(404);
-  response.end();
+
+  const token = request.body['token'];
+  const email = request.body['email'];
+  const now = new Date();
+
+  // Check if link is in links DB
+  const sql = `
+    SELECT * 
+    FROM tokens 
+    WHERE token=$1 
+      AND email=$2
+      AND TO_TIMESTAMP($3) > timestampz
+      And TO_TIMESTAMP($3) < (timestampz + INTERVAL '24 hours')
+    ORDER BY id ASC LIMIT 1`;
+  const values = [token, email, now];
+  pool.query(sql, values).catch(err => {
+    console.log('loginEndpoint returned error');
+    console.log({ error: err });
+  });
+
+  const resultFound = true;
+  if (resultFound) {
```

Comment:
This looks like it's still placeholder code - should this be updated to check the results of the query instead of a `.then` call?

Reply:
The onlyAllowResearchers function is part of another api call that I have not finalized yet. Will fix this part when I work on that api call.

Reply:
Ah cool, then for now I'd just add a comment about this and update it to default to "false" so that if we accidentally ship this it doesn't give access to anyone. :)

Reply:
done!